### PR TITLE
[PM-3683] Remove ipcRenderer from electron-platform-utils

### PR DESCRIPTION
**Pull request by dani-garcia**
## Type of change
```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective
Removed all uses of `ipcRenderer` from electron-platform-utils.service.ts, and moves them to use the ipc preload, getting us step closer to enabling electron sandboxing.

I didn't change it because it's used in a fair few places, but it would seem like a good idea to make sure that `launchUri` only allows opening `https://` and `http://` URIs. It seems dangerous to be allowed to use it with `file://`, but I don't know if anyone depends on that functionality.

### CODE REVIEW

File: apps/desktop/src/services/electron-main-messaging.service.ts [hunk 1]
```diff
@@ -68,6 +78,8 @@ export class ElectronMainMessagingService implements MessagingService {
       alert.show();
     });
 
+    ipcMain.handle("launchUri", async (event, uri) => shell.openExternal(uri));
```
Comment (Hinton) [inline apps/desktop/src/services/electron-main-messaging.service.ts:81]: We have a whitelist in https://github.com/bitwarden/clients/blob/78248db59044d9796ee979be3300585c2e375da3/libs/common/src/vault/models/view/login-uri.view.ts#L8

File: apps/desktop/src/platform/preload.ts [hunk 2]
```diff
@@ -52,8 +66,15 @@ export default {
     });
   },
 
+  clipboardRead: (): Promise<string> => ipcRenderer.invoke("clipboard.read"),
+  clipboardWrite: (message: ClipboardWriteMessage) =>
+    ipcRenderer.invoke("clipboard.write", message),
```
Comment (Hinton) [inline apps/desktop/src/platform/preload.ts:71]: **suggestion:** Have a clipboard module with read/write? I think it better aligns with the considering clipboards as a sub entity.

File: apps/desktop/src/types/biometric-message.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
 export enum BiometricStorageAction {
```
Comment (Hinton) [inline apps/desktop/src/types/biometric-message.ts:1]: **issue:** The name of this enum should be updated. `authenticate` is not a storage action.

**Review by Hinton — CHANGES_REQUESTED** [review]

File: libs/common/src/platform/misc/safe-urls.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  "rdp://", // Legacy RDP URI scheme
+  "ms-rd:", // Preferred RDP URI scheme
+  "chrome://",
+  "iosapp://",
+  "androidapp://",
+];
+
+export class SafeUrls {
+  static canLaunch(uri: string): boolean {
+    if (uri == null || uri === "") {
```
Comment (Hinton) [inline libs/common/src/platform/misc/safe-urls.ts:19]: We have a `Utils.isNullOrWhitespace` that could be used here?

Comment (dani-garcia) [reply libs/common/src/platform/misc/safe-urls.ts]: Good point, changed now

**Review by Hinton — APPROVED** [review]

**Review by LRNcardozoWDF — APPROVED** [review]
LGTM!

**Pull request merged**